app.js

```diff
@@ -1,83 +1,80 @@
 var memory = {
   initial: [],
-  operation: undefined, r: []
+  operation: undefined,
+  latter: []
 }
 
-function operationAdder(div) {
-  memory.operation = this.innerHTML
-}
+document.querySelector('.calculator').addEventListener('click', function(e) {
+
+  //display variable is not working
+  var display = document.querySelector('.calculator-screen').innerHTML
+  var target = e.target || e.srcElement
 
-function numberCounter(div) {
-  if( !memory.operation && (this.innerHTML !== 'AC' || '+/-' || '%' || '.' )) {
-      memory.initial.push(this.innerHTML)
-      document.querySelector('.calculator-screen').innerHTML = memory.initial.join('')
-  }else if( memory.operation && (this.innerHTML !== 'AC' || '+/-' || '%' || '.' )) {
-      memory.latter.push(this.innerHTML)
-      document.querySelector('.calculator-screen').innerHTML = memory.latter.join('')
+  if(memory.operation){
+    memory.current = memory.latter
```

Comment:
memory current seems to make sense but latter/inital seems confusing... 
Yah, these names could probably be better, but I'm not sure what's clearer.